tests/ctests/StatesTest.cpp

```diff
@@ -0,0 +1,488 @@
+  for (int i=0; i < positions.size(); i++) {
+    stateVector.push_back(ale::State(positions[i], velocities[i])); 
+  }
+
+  States statesState(ephemTimes, stateVector, 1);
+  vector<ale::State> states = statesState.getStates();
+  EXPECT_EQ(states.size(), 4);
+  EXPECT_NEAR(states[0].position.x, 4.0, 1e-10);
+  EXPECT_NEAR(states[0].position.y, 1.0, 1e-10);
+  EXPECT_NEAR(states[0].position.z, 4.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.x, -4.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.y, -1.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.z, -4.0, 1e-10);
+  EXPECT_NEAR(states[3].position.x, 7.0, 1e-10);
+  EXPECT_NEAR(states[3].position.y, 4.0, 1e-10);
+  EXPECT_NEAR(states[3].position.z, 1.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.x, -7.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.y, -4.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.z, -1.0, 1e-10);
+}
+
+class TestState : public ::testing::Test {
+protected:
+  States *states;
+
+  // fixtures.h had a lot of overrides before the steup and tearDqwn functions
+  void SetUp() override {
+    states = NULL;
+
+    //define test data values
+    std::vector<double> ephemTimes = {0.0, 1.0, 2.0, 3.0};
+
+    std::vector<ale::Vec3d> positions = {
+      ale::Vec3d(4.0, 1.0, 4.0),
+      ale::Vec3d (5.0, 3.0, 3.5),
+      ale::Vec3d (4.0, 3.5, 2.5),
+      ale::Vec3d (7.0, 4.0, 1.0)};
+
+    std::vector<ale::Vec3d> velocities = {
+      ale::Vec3d(-4.0, -1.0, -4.0),
+      ale::Vec3d(-5.0, -3.0, -3.5),
+      ale::Vec3d(-4.0, -3.5, -2.5),
+      ale::Vec3d(-7.0, -4.0, -1.0)};
+
+    std::vector<ale::State> stateVector;
+    for (int i=0; i < positions.size(); i++) {
+      stateVector.push_back(ale::State(positions[i], velocities[i])); 
+    }
+
+    states = new ale::States(ephemTimes, stateVector);
+  }
+
+  void TearDown() override {
+    if (states) {
+      delete states;
+      states = NULL;
+    }
+  }
+};
+
+TEST_F(TestState, getPosition) {
+  ale::Vec3d linear_position = states->getPosition(0.5, linear);
+  ale::Vec3d spline_position = states->getPosition(2.5, spline);
+
+  EXPECT_NEAR(linear_position.x, 4.5, 1e-10);
+  EXPECT_NEAR(linear_position.y, 2, 1e-10);
+  EXPECT_NEAR(linear_position.z, 3.75, 1e-10);
+  EXPECT_NEAR(spline_position.x, 5.05, 1e-10);
+  EXPECT_NEAR(spline_position.y, 3.7125, 1e-10);
+  EXPECT_NEAR(spline_position.z, 1.7875, 1e-10);
+}
+
+TEST_F(TestState, getVelocity) {
+  ale::Vec3d linear_velocity = states->getVelocity(0.5, linear);
+  ale::Vec3d spline_velocity = states->getVelocity(2.5, spline);
+
+  EXPECT_NEAR(linear_velocity.x, -4.5, 1e-10);
+  EXPECT_NEAR(linear_velocity.y, -2, 1e-10);
+  EXPECT_NEAR(linear_velocity.z, -3.75, 1e-10);
+  EXPECT_NEAR(spline_velocity.x, -5.05, 1e-10);
+  EXPECT_NEAR(spline_velocity.y, -3.7125, 1e-10);
+  EXPECT_NEAR(spline_velocity.z, -1.7875, 1e-10);
+}
+
+// getState() and interpolateState() are tested when testing getPosition and
+// getVelocity, because they are derived from those methods
```

Comment:
I'd like to see a separate getState test. Right now, the implementation of getPosition and getVelocity uses getState, but if that changes then it won't be tested.

Reply:
I think interpolateState doesn't need a separate test.

Reply:
getState is tested right now in the isis comparison. Is that sufficient?